Readability.js

```diff
@@ -245,10 +260,23 @@ Readability.prototype = {
       if (curTitle.split(' ').length < 3)
         curTitle = origTitle.replace(/[^\|\-]*[\|\-](.*)/gi,'$1');
     } else if (curTitle.indexOf(': ') !== -1) {
-      curTitle = origTitle.replace(/.*:(.*)/gi, '$1');
+      // Wheck if we have an heading containing this exact string, so we
+      // could assume it's the full title.
+      var headings = this._concatNodeLists(
+        doc.getElementsByTagName('h1'),
+        doc.getElementsByTagName('h2')
+      );
+      var match = this._someNode(headings, function(heading) {
+        return heading.textContent === curTitle;
+      });
 
-      if (curTitle.split(' ').length < 3)
-        curTitle = origTitle.replace(/[^:]*[:](.*)/gi,'$1');
+      // If we don't, let's extract the title out of the original title string.
+      if (!match) {
+        curTitle = origTitle.replace(/.*:(.*)/gi, '$1');
```

Comment:
`origTitle.replace(/^[^:]*:/, '');` seems more appropriate - don't need to do this more than once, don't need to keep the rest of the string and replace it back in - we can just strip the first bit. Could even do this with `indexOf` and `.substring`

(resurfacing this comment as the diff got updated...)